File: Android/2016-07-24_Kotlin_Android_Extensions/app/src/main/java/com/androidhuman/example/kotlinandroidextensions/MainActivity.java
package com.androidhuman.example.kotlinandroidextensions;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private Button btnExtensionActivity;

    private Button btnExtensionFragment;

    private Button btnExtensionRecyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnExtensionActivity = (Button) findViewById(R.id.btn_activity_main_activity);
        btnExtensionFragment = (Button) findViewById(R.id.btn_activity_main_fragment);
        btnExtensionRecyclerView = (Button) findViewById(R.id.btn_activity_main_recyclerview);

        btnExtensionActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this, ExtensionActivity.class));
            }
        });

        btnExtensionFragment.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this, ExtensionFragmentActivity.class));
            }
        });

        btnExtensionRecyclerView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this, ExtensionRecyclerViewActivity.class));
            }
        });

    }
}

File: Android/2020-11-25_android_view_binding/app/src/main/java/com/androidhuman/example/viewbinding/MainActivity.kt
package com.androidhuman.example.viewbinding

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.androidhuman.example.viewbinding.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.showSecondActivity.setOnClickListener {
            startActivity(Intent(this, SecondActivity::class.java))
        }
    }
}